pkg/metrics/collector/conntrack_test.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2018 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestConntrack(t *testing.T) {
+	input := "entries  searched found new invalid ignore delete delete_list insert insert_failed drop early_drop icmp_error  expect_new expect_create expect_delete search_restart\n" +
+		"000005d2  00000000 00000000 00000000 00000003 00512c38 00000000 00000000 00000000 00000000 00000000 00000000 00000000  00000000 00000000 00000000 00000290\n" +
+		"000005d2  00000000 00000000 00000000 00000002 0047b7ef 00000000 00000000 00000000 00000008 00000005 00000000 00000000  00000000 00000000 00000000 00000255\n" +
+		"000005d2  00000000 00000000 00000000 00000003 0042ab82 00000000 00000000 00000000 00000010 00000000 00000000 00000000  00000000 00000000 00000000 00000245\n" +
+		"000005d2  00000000 00000000 00000000 00000000 00403f1b 00000000 00000000 00000000 00000000 00000020 00000000 00000000  00000000 00000000 00000000 0000026d"
+
+	counters, err := parseConntrackCounters(strings.NewReader(input))
+	if err != nil {
+		t.Errorf("parse conntrack fail: %v", err)
+	}
+
+	if v, ok := counters["insert_failed"]; !ok || v != 24 {
+		t.Errorf("expect insert_failed 24, got %v", v)
+	}
+	if v, ok := counters["drop"]; !ok || v != 37 {
+		t.Errorf("expect drop 37, got %v", v)
+	}
+}
+
+func TestConntrackInvalidInput(t *testing.T) {
+	inputs := []string{
```

Comment:
`t.Parallel()`

use the standard golang testing idiom:

```go
for _, tc := range []struct{
  desc string
  input string
} {
  {
     desc: "empty file",
     input: "",
   },
   ...
} {
   t.Run(tc.desc, func(t testing.T) {
     _, err := parseConntrackCounters(strings.NewReader(tc.input)
    if err == nil {
      t.Errorf("parseConntrackCounters(%q) = _, nil, want error", tc.input)
    }
}
```

Reply:
Updated